### PR TITLE
[master] 3.3.5 -> master cherry-pick

### PR DESCRIPTION
**Pull request by funjoker**
**Changes proposed:**

-  Cherry picking

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -6151,16 +6151,11 @@ void Player::CheckAreaExploreAndOutdoor()
...
-    AreaTableEntry const* areaEntry = sAreaTableStore.LookupEntry(areaId);
-
-    if (sWorld->getBoolConfig(CONFIG_VMAP_INDOOR_CHECK) && !isOutdoor)
+    if (sWorld->getBoolConfig(CONFIG_VMAP_INDOOR_CHECK) && !IsOutdoors())
         RemoveAurasWithAttribute(SPELL_ATTR0_OUTDOORS_ONLY);
 
-    if (!areaId)
-        return;
-
+    uint32 const areaId = GetMap()->GetAreaId(GetPhaseShift(), GetPositionX(), GetPositionY(), GetPositionZ());
```
Comment (funjoker) [inline src/server/game/Entities/Player/Player.cpp:6157]: I don't know if this is correct.

in 335: https://github.com/TrinityCore/TrinityCore/commit/e79c595b695ea5f35b8324efbb7741ccb102bce9#diff-baadebd8cd1117ca48225f316a5ab3fd5fd55b20963394d302341147183db067R6385

File: src/server/scripts/Commands/cs_gobject.cpp [hunk 2]
```diff
@@ -605,14 +605,15 @@ class gobject_commandscript : public CommandScript
...
+        ObjectGuid::LowType spawnId = 0;
         if (strcmp(param1, "guid") == 0)
         {
             char* tail = strtok(nullptr, "");
             char* cValue = handler->extractKeyFromLink(tail, "Hgameobject");
             if (!cValue)
                 return false;
-            ObjectGuid::LowType guidLow = atoull(cValue);
-            GameObjectData const* data = sObjectMgr->GetGameObjectData(guidLow);
+            spawnId = atoul(cValue);
```
Comment (Shauren) [inline src/server/scripts/Commands/cs_gobject.cpp:615]: needs to be `atoull`, like before

File: src/server/game/Spells/Auras/SpellAuras.cpp [hunk 3]
```diff
@@ -2304,14 +2308,14 @@ void UnitAura::FillTargetMap(std::unordered_map<Unit*, uint32>& targets, Unit* c
...
-                        if (!condList || sConditionMgr->IsObjectMeetToConditions(GetUnitOwner(), GetUnitOwner(), *condList))
+                        if (!condList || sConditionMgr->IsObjectMeetToConditions(GetUnitOwner(), caster, *condList))
                             units.push_back(GetUnitOwner());
                         /* fallthrough */
                     case SPELL_EFFECT_APPLY_AREA_AURA_OWNER:
                     {
                         if (Unit* owner = GetUnitOwner()->GetCharmerOrOwner())
                             if (GetUnitOwner()->IsWithinDistInMap(owner, radius))
-                                if (!condList || sConditionMgr->IsObjectMeetToConditions(owner, GetUnitOwner(), *condList))
+                                if (!condList || sConditionMgr->IsObjectMeetToConditions(owner, caster, *condList))
```
Comment (Shauren) [inline src/server/game/Spells/Auras/SpellAuras.cpp:2318]: The same change must be appled to the two cases not present in 3.3.5 branch: SPELL_EFFECT_APPLY_AURA_ON_PET and SPELL_EFFECT_APPLY_AREA_AURA_SUMMONS

File: src/server/game/Entities/Player/Player.cpp (hunk 1 above)
Comment (Shauren) [reply src/server/game/Entities/Player/Player.cpp:6157]: No, it is not, it should be GetAreaId(), just like in 3.3.5

File: src/server/game/Spells/Auras/SpellAuras.cpp (hunk 3 above)
Comment (funjoker) [reply src/server/game/Spells/Auras/SpellAuras.cpp:2318]: is it now correct?

File: src/server/game/Entities/Object/Object.cpp [hunk 4]
```diff
@@ -909,7 +909,7 @@ void WorldObject::setActive(bool on)
     }
 }
 
-void WorldObject::SetVisibilityDistanceOverride(VisibilityDistanceType type)
+    void WorldObject::SetVisibilityDistanceOverride(VisibilityDistanceType type)
```
Comment (Shauren) [inline src/server/game/Entities/Object/Object.cpp:912]: ?

File: src/server/game/DataStores/DB2Stores.cpp [hunk 5]
```diff
@@ -1217,7 +1218,8 @@ uint32 DB2Manager::LoadStores(std::string const& dataPath, LocaleConstant defaul
     for (NamesReservedEntry const* namesReserved : sNamesReservedStore)
     {
         std::wstring name;
-        ASSERT(Utf8toWStr(namesReserved->Name, name));
+        bool conversionResult = Utf8toWStr(namesReserved->Name, name);
+        ASSERT(conversionResult);
```
Comment (Shauren) [inline src/server/game/DataStores/DB2Stores.cpp:1222]: also needed in loop over sNamesReservedLocaleStore

File: src/server/game/Entities/Object/Object.cpp (hunk 4 above)
Comment (funjoker) [reply src/server/game/Entities/Object/Object.cpp:912]: I donno how this happened xD

File: src/server/game/Entities/Creature/Creature.cpp [hunk 6]
```diff
@@ -3232,7 +3232,7 @@ void Creature::FocusTarget(Spell const* focusSpell, WorldObject const* target)
     }
 
     if (noTurnDuringCast)
-        AddUnitState(UNIT_STATE_CANNOT_TURN);
+        AddUnitState(UNIT_STATE_FOCUSING);
```
Comment (Shauren) [inline src/server/game/Entities/Creature/Creature.cpp:3237]: What happened to this commit? Original also has changes in Player.cpp

**Review by Shauren — APPROVED** [review]

**Pull request merged**